src/routes/webhooks.js

```diff
@@ -134,4 +135,44 @@ router.put(
   }
 );
 
+/* This is a PUT request that updates offer trade when paid. */
+router.put(
+  '/offer/order/paid',
+  updateOfferOrderPaidValidator,
+  authenticateApiKey,
+  async (req, res) => {
+    const validator = validateResult(req, res);
+    if (validator.length) {
+      return res.status(400).send(validator);
+    }
+
+    const db = await Database.getInstance(req);
+    const collection = db.collection('orders');
+
+    const order = await collection.findOne({
+      hash: req.body._grinderyTransactionHash,
```

Comment:
Two comments here:
- For me, we are at the completion phase so that the hash to check is not this hash (that corresponds to the hash for the creation of the order) but the `completionHash` that corresponds to the hash of the payment coming from Gordon (either via Zapier or via the UI).
- To be sure and avoid duplicate maybe we can also filter by `chainId`